### PR TITLE
Add typing stubs

### PR DESCRIPTION
**Pull request by bluetech**
There a couple preparatory commits:

-  Mark the wheel universal -- The code works on both Python 2 and 3 so the same wheel can be used for both.

- Use a package layout, src/ and testing/ directories  - needed for the py.typed file

The last commit adds the typing stubs. This is copy/pasted from the stubs of the `py.iniconfig` module in the `py` library, however I removed the `_` prefix from `SectionWrapper` type since it's good to be able to refer to it. I marked its `__init__` as private (there is unfortunately no programmatic way to mark this).

Tested working against pytest's code base (there are a couple legitimate errors I would have to fix there).

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,6 +4,6 @@ envlist=py27,py26,py33,py34,py35
 
 [testenv]
 commands=
-  pytest --junitxml={envlogdir}/junit-{envname}.xml []
+  pytest --junitxml={envlogdir}/junit-{envname}.xml testing []
```
Comment (RonnyPfannschmidt) [inline tox.ini:7]: This is incorrect as it would always enforce all tests, the testdirs ini option should be used

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]
Great setup, im happy to merge after the nitpick is solved

Thanks!

**Comment by bluetech** [comment]
@RonnyPfannschmidt updated to use `testpaths`.

Also added a commit "tox: remove junit generation" which changes the command just to `pytest`, I don't think anything is using the junitxml stuff here. Can remove it though if it is used by something.

File: tox.ini [hunk 2]
```diff
@@ -4,6 +4,11 @@ envlist=py27,py26,py33,py34,py35
 
 [testenv]
 commands=
-  pytest --junitxml={envlogdir}/junit-{envname}.xml []
+  pytest
```
Comment (RonnyPfannschmidt) [inline tox.ini:7]:
```suggestion
  pytest []
```

Comment (bluetech) [reply tox.ini]: I changed it to `{posargs}` now if you don't mind, I was sure the `[]` was some sort of typo...

**Review by RonnyPfannschmidt — APPROVED** [review]

**Comment by RonnyPfannschmidt** [comment]
Thanks!

**Comment by bluetech** [comment]
Thanks for reviewing and merging @RonnyPfannschmidt.

A release would be great once you have time. It would break pytest linting, I will send [this fix](https://github.com/bluetech/pytest/commit/8fd1b1208a587481591a5405c59b17636f5e28ca) then.

**Pull request merged**